src/main/java/dev/binclub/javaception/runtime/ExecutionEngine.java

```
@@ -22,20 +32,696 @@ public static Object invokeMethodObj(InstanceOop instance, MethodInfo method, Ob
+				break;
+			case PUTFIELD:
+				index = ByteUtils.readUnsignedShort(instructions, currentInstruction + 1);
+				ref = (RefInfo) method.owner.runtimeConstantPool[index - 1];
+				((InstanceOop) methodContext.pop()).fields[ref.getID()] = methodContext.pop();
+				break;
+			case INVOKEVIRTUAL:
+			case INVOKESPECIAL:
+				index = ByteUtils.readUnsignedShort(instructions, currentInstruction + 1);
+				ref = (RefInfo) method.owner.runtimeConstantPool[index - 1];
+				MethodInfo targetMethod = ref.getOwner().methods[ref.getID()];
+				Object ret = null;
+				if (targetMethod.descriptor.length == 1) {
+					ret = ExecutionEngine.invokeMethodObj((InstanceOop) methodContext.pop(), targetMethod);
+				} else {
+					Object[] params = new Object[targetMethod.descriptor.length - 1];
+					for (int param = targetMethod.descriptor.length - 1; param > -1; param--) {
+						params[param] = methodContext.pop();
+					}
+					ret = ExecutionEngine.invokeMethodObj((InstanceOop) methodContext.pop(), targetMethod, params);
+				}
+				if (targetMethod.descriptor[targetMethod.descriptor.length - 1] != PrimitiveType.VOID) {
+					methodContext.push(ret);
+				}
+				break;
+			case INVOKESTATIC:
+				index = ByteUtils.readUnsignedShort(instructions, currentInstruction + 1);
+				ref = (RefInfo) method.owner.runtimeConstantPool[index - 1];
+				targetMethod = ref.getOwner().methods[ref.getID()];
+				if (targetMethod.descriptor.length == 1) {
+					ret = ExecutionEngine.invokeMethodObj(null, targetMethod);
+				} else {
+					Object[] params = new Object[targetMethod.descriptor.length];
+					for (int param = targetMethod.descriptor.length - 1; param > -1; param--) {
+						params[param] = methodContext.pop();
+					}
+					ret = ExecutionEngine.invokeMethodObj(null, targetMethod, params);
+				}
+				if (targetMethod.descriptor[targetMethod.descriptor.length - 1] != PrimitiveType.VOID) {
+					methodContext.push(ret);
+				}
+				break;
+			case INVOKEINTERFACE:
+				break;
```

Should probably throw an UnsupportedOperationException here